File: models/datasource.go
package models

import (
	"time"
)

// DataSource 数据源模型
type DataSource struct {
	ID           string    `gorm:"primaryKey;size:36" json:"id"`
	Name         string    `gorm:"size:100;not null;index" json:"name"`
	Type         string    `gorm:"size:20;not null;index" json:"type"` // mysql/elasticsearch
	Host         string    `gorm:"size:255;not null" json:"host"`
	Port         int       `gorm:"not null" json:"port"`
	Username     string    `gorm:"size:100;not null" json:"username"`
	Password     string    `gorm:"size:255;not null" json:"password"` // 加密存储
	DatabaseName string    `gorm:"size:100" json:"database_name"`     // MySQL专用
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

// TableName 指定表名
func (DataSource) TableName() string {
	return "data_sources"
}

File: datatrace-ui/src/types/datasource.ts
export interface CredentialOption {
  id: string
  name: string
  username: string
  description?: string
}

export interface DataSourceRecord {
  id: string
  name: string
  type: 'mysql' | 'elasticsearch' | string
  host: string
  port: number
  username?: string
  database_name?: string
  credential_id?: string | null
}

export interface DataSourceFormData {
  id?: string
  name: string
  type: 'mysql' | 'elasticsearch'
  host: string
  port: number | null
  auth_type: 'manual' | 'credential'
  credential_id: string
  username: string
  password: string
  database_name: string
}

export interface DataSourcePayload {
  name: string
  type: string
  host: string
  port: number
  credential_id?: string
  username?: string
  password?: string
  database_name?: string
}

export interface TestConnectionResult {
  success: boolean
  version: string
  message: string
}

export interface DataSourceHealthStatus {
  id: string
  status: 'testing' | 'success' | 'failed' | string
  message: string
  timestamp: string
}
